### PR TITLE
add setting for default results storage block

### PR DESCRIPTION
**Pull request by zzstoatzz**
This PR is a continuation of #10404, graciously begun by @manaw. It aims to add a setting that makes it easy to specify a default storage block for results.

## Original User PR body

Closes https://github.com/PrefectHQ/prefect/issues/9226

Add a setting `PREFECT_DEFAULT_RESULT_STORAGE_BLOCK` that accepts the block slug. Results would be saved to that block by default unless a different storage path or block is explicitly defined in the flow code.

## Example
Dockerfile:
```Dockerfile
FROM prefecthq/prefect:2.10.3-python3.10

COPY requirements.txt .

RUN pip install -r requirements.txt

ENV PREFECT_RESULTS_PERSIST_BY_DEFAULT=true
ENV PREFECT_DEFAULT_RESULT_STORAGE_BLOCK=s3/my-prefect-results
```
### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: tests/results/test_result_factory.py [hunk 1]
```diff
@@ -84,7 +90,77 @@ def foo():
...
+    ):
+        result = await foo()
+
+    assert result == {"foo": "bar"}
+    local_storage = await LocalFileSystem.load("my-result-storage")
+    result_bytes = await local_storage.read_path("~/.prefect/results/my-result.pkl")
+    saved_python_result = pickle.loads(
+        base64.b64decode(PersistedResultBlob.parse_raw(result_bytes).data)
+    )
+    assert saved_python_result == {"foo": "bar"}
```
Comment (zzstoatzz) [inline tests/results/test_result_factory.py:142]: is there a better way to go from `storage_result_key` -> `saved_python_result`?

File: src/prefect/tasks.py [hunk 2]
```diff
@@ -303,10 +304,14 @@ def __init__(
         if retry_jitter_factor is not None and retry_jitter_factor < 0:
             raise ValueError("`retry_jitter_factor` must be >= 0.")
 
-        self.retry_jitter_factor = retry_jitter_factor
+        self.result_storage = (
+            storage
+            if (storage := PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value()) is not None
+            else result_storage
+        )
```
Comment (desertaxle) [inline src/prefect/tasks.py:311]: I think this is redundant with the change to `get_default_result_storage` and could be removed.

File: tests/results/test_result_factory.py [hunk 3]
```diff
@@ -84,7 +90,59 @@ def foo():
     assert result_factory.persist_result is True
 
 
-def test_roto_flow_can_opt_out_when_persist_result_default_is_overriden_by_setting():
+async def test_root_flow_default_remote_storage():
```
[the remark this answers is not in the document]
Comment (desertaxle) [reply tests/results/test_result_factory.py:93]: Yeah, I agree these tests should be moved to `test_flow_results.py` instead.

File: tests/results/test_result_factory.py (hunk 1 above)
Comment (zzstoatzz) [reply tests/results/test_result_factory.py:142]: > if you initialize a `PersistedResult`

it seems that you need to provide the actual python `obj` in order to call `PersistedResult.create`, which I was thinking should be done on my behalf when I run the flow with the new enabled setting

after the running the flow, I'd like to prove that the object my task returned is in fact stored where the result storage setting + my result storage key says it is - so I'm not sure I can use `PersistedResult` here, does that track?

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**